File: unrar.pl
#!/usr/bin/perl

use strict;

use warnings;

# install: $ sudo CPAN Path::Class
use Path::Class;

# install: $ sudo CPAN File::chdir
use File::chdir;

#

# Specify starting directory to search for files
my $dir = dir('/home/user/Input');

$CWD = $dir;

print "Working directory: " . $CWD . "\n\n";

# Iterate over the content of $dir
while (my $file = $dir->next) {
    
    # See if it is a directory
    if ( $file->is_dir() ) {

    	# Change working dir to current dir
    	local $CWD = $file;

    	# Execute bash program by calling script
    	system("unrar e *.rar /home/user/Output");

    }

next } # End while loop

exit 0;

File: unrar.py
 #!/bin/python

# import modules
import os
import sys

# set cwd
currentDir = "/home/user/input/"

# parse each item in child dir
for item in os.listdir(currentDir):
    # if item is dir, change to it
    if os.path.isdir(item):
        os.chdir(item)
        # extract rar files
        cmd = "unrar e *.rar /home/user/output"
        os.system(cmd)
